pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -625,5 +639,13 @@ def setstate(k: str, item: dict) -> Union[ResultMetric, ResultMetricCollection]:
     def state_dict(self) -> dict:
         return self.__getstate__()
 
-    def load_state_dict(self, state_dict: dict, map_location: Optional[Union[str, torch.device]] = None) -> None:
-        self.__setstate__(state_dict, map_location=map_location)
+    def load_state_dict(
+        self,
+        state_dict: dict,
+        map_location: Optional[Union[str, torch.device]] = None,
+        sync_fn: Optional[Callable] = None,
+        should_reset: bool = False,
+    ) -> None:
+        self.__setstate__(state_dict, map_location=map_location, sync_fn=sync_fn)
+        if should_reset:
+            self.reset()
```

Comment:
Doesn't seem right to me that reset is part of `load_state_dict`

Reply:
Moved out !